Include/cpython/optimizer.h

```diff
@@ -30,16 +30,52 @@ typedef struct {
     PyCodeObject *code;  // Weak (NULL if no corresponding ENTER_EXECUTOR).
 } _PyVMData;
 
+#define UOP_FORMAT_TARGET 0
+#define UOP_FORMAT_EXIT 1
+#define UOP_FORMAT_JUMP 2
+#define UOP_FORMAT_UNUSED 3
+
 typedef struct {
-    uint16_t opcode;
+    uint16_t opcode:14;
+    uint16_t format:2;
```

Comment:
It may never correct to *execute* an instruction with the wrong format, but this is primarily a format for optimization.
The "target" format is the correct format for *all* instructions during optimization.

If we care about tier 2 interpreter performance, we will want a different format for execution anyway.
What we could do is use different formats before and after `prepare_for_execution`, but I'd rather leave than for another PR.

What's wrong with bitfields?